Lib/tkinter/test/test_tkinter/test_misc.py

```diff
@@ -48,6 +48,39 @@ def test_tk_setPalette(self):
                 '^must specify a background color$',
                 root.tk_setPalette, highlightColor='blue')
 
+    def test_after_cancel(self):
+        root = self.root
+        timer1 = root.after(5000, lambda: 'break')
+        idle1 = root.after_idle(lambda: 'break')
+
+        # No value for id raises a ValueError.
+        with self.assertRaises(ValueError):
+            root.after_cancel(None)
+
+        # A non-existent id raises a TclError, which is caught in after_cancel.
+        with self.assertRaises(tkinter.TclError):
+            root.tk.call('after', 'info', 'spam')
+        root.after_cancel('spam')
+
+        # Cancel timer event.
+        (script, _) = root.tk.splitlist(root.tk.call('after', 'info', timer1))
+        self.assertIn(script, root._tclCommands)
```

Comment:
`_tclCommands` is an implementation detail. It would be better to not use it in tests. Otherwise changing the implementation (it already was changed in the past) will break tests.

But you can test a side effect of the script by calling `root.tk.call(script)`.

Reply:
Thank you for this suggestion.  I wasn't quite sure how to test that the `script` still existed after the cancel.